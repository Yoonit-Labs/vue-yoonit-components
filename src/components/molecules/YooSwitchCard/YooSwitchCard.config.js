module.exports = {
  size: {
    options: [
      'small',
      'medium',
      'large'
    ]
  },
  textPosition: {
    options: [
      'left',
      'right'
    ]
  }
}
