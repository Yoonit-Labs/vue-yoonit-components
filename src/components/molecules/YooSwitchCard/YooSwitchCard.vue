<template lang="pug">
  YooFlexLayout.m__t--s.m__b--s.m__r--l.m__l--l.p__t--l.p__r--l.p__b--l.p__l--l(
    gap="16px"
    :class="['yoo-switch-card', takeChecked]"
    ref="switch_card"
  )
    YooCheckButton(
      :class="['yoo-switch-card__checkbox']"
      :text="text"
      :size="size"
      :switchCard="true"
      :initialValue="isChecked"
      :textPosition="textPosition"
      @response="doCheck"
    )
</template>

<script>
import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'
import YooCheckButton from '@/components/atoms/YooCheckButton/YooCheckButton.vue'
import PropsConfig from '@/components/molecules/YooSwitchCard/YooSwitchCard.config'

export default {
  name: 'YooSwitchCard',
  props: {
    text: {
      type: String,
      required: true,
      default: ''
    },
    textPosition: {
      type: String,
      required: false,
      default: 'right',
      validator: value => PropsConfig.textPosition.options.includes(value)
    },
    checked: {
      type: Boolean,
      required: false,
      default: false
    },
    size: {
      type: String,
      required: false,
      default: 'medium',
      validator: value => PropsConfig.size.options.includes(value)
    }
  },
  components: {
    YooCheckButton,
    YooFlexLayout
  },
  data: () => ({
    isChecked: ''
  }),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {
    this.isChecked = this.checked
  },
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {
    takeChecked () {
      if (this.isChecked) {
        return 'yoo-switch-card--active'
      } else {
        return ''
      }
    }
  },
  methods: {
    doCheck (e) {
      this.isChecked = e
      this.$emit('response', e)
    }
  },
  filters: {},
  watch: {
    checked (value) {
      this.isChecked = value
    }
  }
}
</script>

<style src="./YooSwitchCard.sass" lang="sass" scoped></style>
