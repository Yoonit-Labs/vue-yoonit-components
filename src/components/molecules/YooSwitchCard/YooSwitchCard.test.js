import { shallowMount } from '@vue/test-utils'
import YooSwitchCard from './YooSwitchCard.vue'
// import PropsConfig from './YooSwitchCard.config'

const classBlock = 'yoo-switch-card'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooSwitchCard, {
    slots: { default: SlotText },
    propsData: {
      text: SlotText
    }
  })
}

describe('YooSwitchCard Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo-switch-card')).toBe(true)
    expect(wrapper.find('.yoo-switch-card__checkbox').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('checked', () => {
      it('Has a valid default value', () => {
        expect(YooSwitchCard.props.checked.default).toBe(false)
      })
      it(`Includes checked class: .${classBlock}--active when checked`, async () => {
        await wrapper.setProps({ checked: true })
        expect(wrapper.find(`.${classBlock}--active`).exists()).toBe(true)
      })
      it(`Does not include checked class:.${classBlock}--active when not checked`, async () => {
        await wrapper.setProps({ checked: false })
        expect(wrapper.find(`.${classBlock}--active`).exists()).toBe(false)
      })
    })
  }) // describe Props

  /* describe('textPosition', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.textPosition.options.includes(YooSwitchCard.props.textPosition.default)).toBe(true)
      })
      PropsConfig.textPosition.options.forEach(textPosition => {
        it('Includes checked class: yoo-check--text-right', async () => {
          wrapper = shallowMount(YooSwitchCard, { propsData: { text: 'lala', textPosition } })
          await wrapper.vm.$nextTick()
          expect(wrapper.find('.yoo-check--text-right').exists()).toBe(true)
        })
      })
    })
  }) */

  describe('Events', () => {
    describe('doCheck', () => {
      it('Emits doCheck Event', async () => {
        await wrapper.find(`.${classBlock}__checkbox`).vm.$emit('response')
        expect(wrapper.emitted()).toHaveProperty('response')
      })
    })
  }) // describe Events
})
