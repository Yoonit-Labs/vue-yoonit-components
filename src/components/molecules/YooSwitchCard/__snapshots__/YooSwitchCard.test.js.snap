// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSwitchCard Component Matches Snapshot 1`] = `
<yooflexlayout-stub flexdirection="row" flexwrap="nowrap" justifycontent="center" alignitems="stretch" aligncontent="stretch" gap="16px" class="m__t--s m__b--s m__r--l m__l--l p__t--l p__r--l p__b--l p__l--l yoo-switch-card">
  <yoocheckbutton-stub text="Default Slot Text" size="medium" textposition="right" switchcard="true" class="yoo-switch-card__checkbox"></yoocheckbutton-stub>
</yooflexlayout-stub>
`;
